db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/FromBaseTable.java

```diff
@@ -1633,6 +2369,35 @@ public void bindResultColumns(FromList fromListParam)
         /* Nothing to do, since RCL bound in bindNonVTITables() */
     }
 
+    private void buildRowIdColumn(boolean isRowId) throws StandardException {
+        String colName = isRowId ? ROWID : BASEROWID;
```

Comment:
What's the difference between ROWID and BASEROWID?

Reply:
ROWID doesn't always retrieve the base table row id.  If we do an IndexScan without IndexLookup, it is the row id of the index row.
BASEROWID always retrieves the base table row id.